rep-0149.rst

```diff
@@ -0,0 +1,973 @@
+specified in REP-0127 [1]_ and REP-0140 [2]_.
+It is relevant for packages using the catkin as well as the ament build system.
+
+.. raw:: html
+
+  </font>
+
+
+Motivation
+==========
+
+.. raw:: html
+
+  <font color="red">
+
+This REP covers three separate topics which are described in the following
+subsections.
+
+Dependency groups
+-----------------
+
+Several ROS packages aggregate dependencies of a specific type, e.g.
+`message_generation` in ROS 1 or `rmw_implementation` in ROS 2.
+They explicitly list the dependencies which are being used during the release
+process.
+But when building from source they need to be updated to add custom message
+generators or rmw implementations.
+
+The goal is to allow specifying this kind of group dependency in a way that a
+from source build can process the custom packages accordingly to make them
+available to the package even though the package doesn't list an explicit
+dependency on the custom package.
+
+ABI version on the package level
+--------------------------------
+
+When building binary packages on the ROS buildfarm there is no assumption
+about ABI compatibility.
+When a new Debian package has been built all downstream packages are being
+rebuild too in order to ensure they still work together.
+This makes e.g. releasing a core package like `catkin` or `roscpp`
+particularly costly.
+Due to that overhead changes which should be released rather sooner than
+later, e.g. a documentation fix, simple bugfix in a `.cpp` file, are being
+hold back to group them with other changes to make the rebuild effort
+worthwhile.
+
+In order to allow package releases without enforcing a rebuild of all
+downstream packages a package should be able to declare if such a rebuild
+is necessary or not.
+
+Dependencies across ROS 1 and ROS 2
+-----------------------------------
+
+In order for a ROS package to work with ROS 1 and ROS 2 from a single source
+the dependencies listed in the package manifest are a problem.
+E.g. in ROS 1 a package needs to depend on `roscpp` where as in ROS 2 it needs
+to depend on `rclcpp`.
+Currently the manifest doesn't support this use case.
+
+.. raw:: html
+
+  </font>
+
+
+Rationale
+=========
+
+.. raw:: html
+
+  <font color="green">
+
+REP-0127 [1]_ and REP-0140 [1]_ provide the ``package.xml`` design rationale
+for format one and two, which is not repeated here.
+
+.. raw:: html
+
+  </font>
+
+.. raw:: html
+
+  <font color="red">
+
+Group dependencies
+------------------
+
+While "normal" dependencies are specified in the package which expresses a
+dependency on another package this is not adequate for the targeted goal.
+Instead packages need to be able to declare that they are a dependency of
+another package.
+
+For that kind of "reverse" dependency a package could use the name of the other
+package.
+But that approach would fail if the other package gets forked under a different
+name since the reverse dependency would still reference the original package
+name.
```

Comment:
I know what you mean here, but I have a hard time believing this is enough for others to understand the situation you're talking about. Without the example of the package that depends on all message packages (e.g. `ros1_bridge`), it's hard to imagine. So maybe you should describe that scenario to make it clearer why a reverse dependency makes it hard to change the name of the reverse dependent.